phac-nml/irida-next
# Pathogen: Tabs

## What does this PR do and why?
This PR adds the `tab_panel` to the Pathogen components library. Tabs can be either `default` or `underline` type (screenshots below).

Future PRs will handle implementing the option for an icon to each tab and handling adding 'right-side' options of the panel (search bar, buttons, etc.).

## Screenshots or screen recordings
Default panel:
![image](https://github.com/user-attachments/assets/f36d8e55-ac84-4ea1-919b-c835769f8866)

![image](https://github.com/user-attachments/assets/dee5d07d-8380-4e27-a2ec-97bd67280ecd)

![image](https://github.com/user-attachments/assets/7aba2e62-b8a5-49e4-ae30-e1cde5880cee)

![image](https://github.com/user-attachments/assets/9d7a2ddb-6615-4cc9-a417-f38d2bb38b5e)

Underline panel:

![image](https://github.com/user-attachments/assets/9ee0dd4f-262b-4580-9c81-d0c1aa7a3ccf)

![image](https://github.com/user-attachments/assets/589f2ef3-743f-4d14-b26f-824616c6f7f6)

![image](https://github.com/user-attachments/assets/658355c2-b263-47b3-bf17-c09c476cba27)

![image](https://github.com/user-attachments/assets/26bccece-483a-4bb1-816d-f7aa94a1f792)

## How to set up and validate locally
1. Navigate to the `lookbook` and verify the [previews](http://localhost:3000/rails/lookbook/inspect/pathogen_tabs_panel/default) look as expected
2. Tests pass locally

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
